rust/signed_doc/src/metadata/supported_field.rs

```diff
@@ -0,0 +1,285 @@
+//! Catalyst Signed Document unified metadata field.
+
+#[cfg(test)]
+use std::convert::Infallible;
+use std::{
+    cmp,
+    ops::{Deref, DerefMut},
+};
+
+use catalyst_types::uuid::UuidV7;
+use strum::{EnumDiscriminants, EnumTryAs, IntoDiscriminant as _};
+
+use crate::{
+    metadata::{MetadataDecodeContext, MetadataEncodeContext},
+    ContentEncoding, ContentType, DocType, DocumentRef, Section,
+};
+
+/// COSE label. May be either a signed integer or a string.
+#[derive(Copy, Clone, Eq, PartialEq)]
+enum Label<'a> {
+    /// Integer label fitting into `0..=23`.
+    U8(u8),
+    /// Text label.
+    Str(&'a str),
+}
+
+impl<C> minicbor::Encode<C> for Label<'_> {
+    fn encode<W: minicbor::encode::Write>(
+        &self, e: &mut minicbor::Encoder<W>, _: &mut C,
+    ) -> Result<(), minicbor::encode::Error<W::Error>> {
+        match self {
+            &Label::U8(u) => e.u8(u),
+            Label::Str(s) => e.str(s),
+        }?
+        .ok()
+    }
+}
+
+impl<'a, C> minicbor::Decode<'a, C> for Label<'a> {
+    fn decode(d: &mut minicbor::Decoder<'a>, _: &mut C) -> Result<Self, minicbor::decode::Error> {
+        match d.datatype()? {
```

Comment:
COSE isn't strictly limited to 8 bits for label.  But for practical purposes it is because its < 7